File: src/Database/Model/ProductionOrder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace AasDemoapp.Database.Model
{
    public class ProductionOrder : ISoftDelete
    {
        public long? Id { get; set; }
        public long ConfiguredProductId { get; set; }
        public ConfiguredProduct ConfiguredProduct { get; set; } = null!;
        public int Anzahl { get; set; } = 1;
        public long? AddressId { get; set; }
        public Address? Address { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public DateTime? UpdatedAt { get; set; }
        public DateTime? FertigstellungsDatum { get; set; }
        public bool ProduktionAbgeschlossen { get; set; } = false;
        public bool Versandt { get; set; } = false;
        public DateTime? VersandDatum { get; set; }
        public bool IsDeleted { get; set; }
        public DateTime? DeletedAt { get; set; }
    }
}

File: src/ClientApp/src/app/model/production-order.ts
// ProductionOrder DTOs
export interface ProductionOrder {
  id?: number;
  configuredProductId: number;
  configuredProductName: string;
  anzahl: number;
  address?: Address;
  createdAt: Date;
  updatedAt?: Date;
  fertigstellungsDatum?: Date;
  produktionAbgeschlossen: boolean;
  versandt: boolean;
  versandDatum?: Date;
}

export interface Address {
  id?: number;
  name?: string;
  vorname?: string;
  strasse?: string;
  plz?: string;
  ort?: string;
  land?: string;
  lat?: number;
  long?: number;
}

export interface CreateProductionOrder {
  configuredProductId: number;
  anzahl: number;
  address?: Address;
}

export interface ProductionOrderResponse {
  success: boolean;
  message: string;
  productionOrder?: ProductionOrder;
  error?: string;
}

// Enums für die Filterung
export enum ProductionOrderStatus {
  PENDING_PRODUCTION = 'PENDING_PRODUCTION',
  READY_FOR_SHIPPING = 'READY_FOR_SHIPPING',
  COMPLETED = 'COMPLETED',
}
